intent_classification/cnn/train_with_pretrained_embeddings.py

```diff
@@ -0,0 +1,100 @@
+import tensorflow as tf
+import math
+import sys
+sys.path.append("../")
+import atis_data
+
+FLAGS = tf.app.flags.FLAGS
+tf.app.flags.DEFINE_string("data_dir", '../../ATIS', "Data directory")
+tf.app.flags.DEFINE_integer("in_vocab_size", 10000, "max vocab Size.")
+tf.app.flags.DEFINE_string("max_in_seq_len", 30, "max in seq length")
+tf.app.flags.DEFINE_integer("max_data_size", 10000, "max training data size")
+tf.app.flags.DEFINE_integer("batch_size", 100, "batch size")
+tf.app.flags.DEFINE_integer("iterations", 10000, "number of iterations")
+tf.app.flags.DEFINE_integer("embedding_size", 300, "size of embedding")
+
+def train():
+	atis = atis_data.AtisData(FLAGS.data_dir, FLAGS.in_vocab_size,
+			 FLAGS.max_in_seq_len, FLAGS.max_data_size, FLAGS.embedding_size)
+
+	# define parameters of cnn
+	num_filters = 32
+	filter_sizes = [2, 3, 4]
+	num_filters_total = num_filters * len(filter_sizes)
+	dropout_keep_prob = 0.4
+	normal_initializer = tf.random_normal_initializer(stddev=0.1)
+
+	#get data
+	num_classes = atis.get_number_of_labels()
+	l2_loss = tf.constant(0.0)
+
+	train_x = tf.placeholder(tf.float32, [None, FLAGS.max_in_seq_len, FLAGS.embedding_size])
```

Comment:
Ah, this is what we talked about today, right?

The `[batch_size, sentence_length, embedd_size]` dimension  - which is of the `W`, not the output.

Reply:
yep.